invenio_search/ext.py

```diff
@@ -452,24 +452,34 @@ def update_mapping(self, index):
             else:
                 raise NotAllowedMappingUpdate(list(changes))
 
-    def _replace_prefix(self, template_path, body):
+    def _replace_prefix(self, template_path, body, enforce_prefix):
         """Replace index prefix in template request body."""
         pattern = "__SEARCH_INDEX_PREFIX__"
 
         prefix = self.app.config["SEARCH_INDEX_PREFIX"] or ""
         if prefix:
-            assert pattern in body, (
-                "You are using the prefix `{0}`, but the template `{1}` does not "
-                "contain the pattern `{2}`."
+            message = (
+                "You are using the prefix `{0}`, but the template `{1}` does not contain the pattern `{2}`."
             ).format(prefix, template_path, pattern)
```

Comment:
nit: we could also use Python f-strings here.

Reply:
Done, thanks!